### PR TITLE
fixed logout callback and type rules for change_password action

### PR DESCRIPTION
**Pull request by mirceaalexandru**

### CODE REVIEW

File: user.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ function conditionalExtend(user, args) {
 module.exports = function user(options) {
   var seneca = this
 
+  var user_coll = 'sys/user'
```
Comment (rjrodger) [inline user.js]: "collection" is not a seneca term - use user_canon etc

File: user.js [hunk 2]
```diff
@@ -261,15 +278,17 @@ module.exports = function user(options) {
   })
 
 
+  function cmd_load_user( args, cb ) {
```
Comment (rjrodger) [inline user.js]: use "done" instead of "cb" - make this consistent

File: user.js [hunk 3]
```diff
@@ -588,12 +607,12 @@ module.exports = function user(options) {
 
       login.token = login.id$, // DEPRECATED
 
-      login.save$( function( err, login ){
-        if( err ) return done(err);
+        login.save$( function( err, login ){
```
Comment (rjrodger) [inline user.js]: are these indents correct?

File: user.js [hunk 4]
```diff
@@ -684,7 +703,7 @@ module.exports = function user(options) {
       if( err ) return done(err);
 
       if( !login ) {
-        return {ok:true}
+        return done(null, {ok:true})
```
Comment (rjrodger) [inline user.js]: ha ha ha - I'm so stupid :)

File: user.js [hunk 5]
```diff
@@ -195,57 +199,70 @@ module.exports = function user(options) {
...
+    user:{required$:true,entity$:user_coll}, // sys/user entity
   }, cmd_clean )
 
 
+  // ### Lod user based on a query. Required by external seneca plugins such as seneca-auth
+  // Pattern: _**role**:user, **cmd**:load_user_
+  // Returns an entity based on a query
+  seneca.add(
+    {
+      role:role,
```
Comment (rjrodger) [inline user.js]: change pattern to role:user,get:user

File: user.js [hunk 6]
```diff
@@ -195,57 +199,70 @@ module.exports = function user(options) {
...
+      cmd:'load_user',
+      q:{object$:true,required$:true}
```
Comment (rjrodger) [inline user.js]: using q in this way exposes implementation details
instead, merge all remaining properties into a query:

q = _.omit( args, ['role','get'] )

**Comment by mihaidma** [comment]
Please update the links in the readme from `rjrodger` to `senecajs`, some links are broken.
There are also some linting issues.

It looks ok, can be merged.

**Comment by mihaidma** [comment]
:+1:

**Pull request merged**